api.go

```diff
@@ -31,28 +29,18 @@ type BrokerCredentials struct {
 }
 
 func New(serviceBroker ServiceBroker, logger lager.Logger, brokerCredentials BrokerCredentials) http.Handler {
-	authMiddleware := auth.NewWrapper(brokerCredentials.Username, brokerCredentials.Password).Wrap
-	return NewWithCustomAuth(serviceBroker, logger, authMiddleware)
+	return NewWithOptions(serviceBroker, logger, WithBrokerCredentials(brokerCredentials))
 }
 
 func NewWithCustomAuth(serviceBroker ServiceBroker, logger lager.Logger, authMiddleware mux.MiddlewareFunc) http.Handler {
-	router := mux.NewRouter()
-
-	AttachRoutes(router, serviceBroker, logger)
-
-	apiVersionMiddleware := middlewares.APIVersionMiddleware{LoggerFactory: logger}
-
-	router.Use(middlewares.AddCorrelationIDToContext)
-	router.Use(authMiddleware)
-	router.Use(middlewares.AddOriginatingIdentityToContext)
-	router.Use(apiVersionMiddleware.ValidateAPIVersionHdr)
-	router.Use(middlewares.AddInfoLocationToContext)
-	router.Use(middlewares.AddRequestIdentityToContext)
-
-	return router
+	return NewWithOptions(serviceBroker, logger, WithCustomAuth(authMiddleware))
 }
 
 func AttachRoutes(router *mux.Router, serviceBroker ServiceBroker, logger lager.Logger) {
+	NewWithOptions(serviceBroker, logger, WithRouter(router))
```

Comment:
This should be calling to just `attachRoutes` rather then `NewWithOptions`

Reply:
I think it is fine, NewWithOptions just creates a config (router, logger, and a flag for customRouter) and sets the logger and the router that has been passed as parameters in the config. The setting of the default middleware will be ignored because it was set up with a custom router. So basically it will just call attachRoutes on the parameters passed.

Reply:
Given that this is a public method that users of the library rely on I think we should be very clear with its intended behaviour. Any changes in  `NewWithOptions` in the future can have unintended consequences for this method. I think calling `NewWithOptions` just makes it less expressive and more prone to unintended consequences.